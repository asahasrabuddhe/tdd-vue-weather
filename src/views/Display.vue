<template>
    <div class="level-item" v-if="weather.main !== undefined">
        <div class="weather">
            <p class="subtitle is-5 city">The weather in <strong>{{weather.name}}</strong> is:</p>
            <br/>
            <p class="temp"><strong>Temperature:</strong> {{weather.main.temp}}&deg;C</p>
            <p class="humidity"><strong>Humidity:</strong> {{weather.main.humidity}} %</p>
            <p class="wind"><strong>Wind Speed:</strong> {{weather.wind.speed}} m/s</p>
        </div>
    </div>
</template>

<script>
    export default {
        name: "Display",
        props: {
            weather: {
                type: Object,
                required: true,
                default: null
            }
        }
    }
</script>

<style scoped>
    .weather {
        text-align: left
    }
</style>
