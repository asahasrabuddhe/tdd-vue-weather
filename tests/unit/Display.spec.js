import Display from '@/views/Display';
import {shallowMount} from '@vue/test-utils';
import weatherData from './data/weather'

describe('Display', () => {
    let props

    const build = () => {
        const wrapper = shallowMount(Display, {
            propsData: props,
        })

        return {
            wrapper,
            city: () => wrapper.find('.weather .city'),
            temp: () => wrapper.find('.weather .temp'),
            humidity: () => wrapper.find('.weather .humidity'),
            wind: () => wrapper.find('.weather .wind'),
        }
    }

    beforeEach(() => {
        props = {
            weather: weatherData.data
        }
    })

    it('renders the component properly', () => {
        const { wrapper } = build()

        expect(wrapper.html()).toMatchSnapshot()
    })

    it('renders its contents properly', () => {
        const { city, temp, humidity, wind } = build()

        expect(city().exists()).toBeTruthy()
        expect(city().text()).toBe(`The weather in ${props.weather.name} is:`)

        expect(temp().exists()).toBeTruthy()
        expect(temp().text()).toBe(`Temperature: ${props.weather.main.temp.toString()}Â°C`)

        expect(humidity().exists()).toBeTruthy()
        expect(humidity().text()).toBe(`Humidity: ${props.weather.main.humidity.toString()} %`)

        expect(wind().exists()).toBeTruthy()
        expect(wind().text()).toBe(`Wind Speed: ${props.weather.wind.speed} m/s`)
    })
})
