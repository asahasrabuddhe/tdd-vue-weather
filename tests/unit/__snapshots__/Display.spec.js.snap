// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders the component properly 1`] = `
<div class="level-item">
  <div class="weather">
    <p class="subtitle is-5 city">The weather in <strong>Pune</strong> is:</p> <br>
    <p class="temp"><strong>Temperature:</strong> 29.87Â°C</p>
    <p class="humidity"><strong>Humidity:</strong> 39 %</p>
    <p class="wind"><strong>Wind Speed:</strong> 4.762 m/s</p>
  </div>
</div>
`;
